src/test/java/net/minestom/server/collision/PlayerBlockTouchTickIntegrationTest.java

```diff
@@ -0,0 +1,313 @@
+package net.minestom.server.collision;
+
+import net.minestom.server.coordinate.Point;
+import net.minestom.server.coordinate.Pos;
+import net.minestom.server.coordinate.Vec;
+import net.minestom.server.instance.block.Block;
+import net.minestom.server.instance.block.BlockHandler;
+import net.minestom.server.utils.Direction;
+import net.minestom.server.utils.NamespaceID;
+import net.minestom.testing.Env;
+import net.minestom.testing.EnvTest;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+// Separate because of how movement is calculated between players and entities.
+// We should expect this bad behavior from the player but not entities. unless the physics engine has gets a controller input vector.
+@EnvTest
+public class PlayerBlockTouchTickIntegrationTest {
```

Comment:
Nitpick: Remove public for junit5.
Ref: https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-classes-and-methods